### PR TITLE
feat(migration): Add `visiblity` to savedsearch model

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This does two things

 - Adds the `visibility` text column with a default of `owner`

 - Updates create calls to SavedSearch to set the correct intermittent value for visibility. Since right now all saved searches are organization level, newly created saved searches should be `Visibility.ORGANIZATION`, and pinned searches are `Visibility.OWNER_PINNED`.

The second step is neccissary since immediately after this we'll be running a migration that marks all existing saved searches as `Visibility.ORGANIZATION` and pinned searches as `Visibility.OWNER_PINNED`.

Requires GH-40878

### CODE REVIEW

**Comment by evanpurkhiser** [comment]
I'm aware this is missing the migration_lock.txt as well. I'll add it in once dependencies are merged.

**Review by wedamija — APPROVED** [review]
Looks good although I think you need to rebase your migration

File: src/sentry/migrations/0338_add_saved_search_visibility.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not. Generally
+    # we don't want to run in a transaction here, since for long running operations like data
+    # back-fills this results in us locking an increasing number of rows until we finally commit.
+    atomic = False
+
+    dependencies = [
+        ("sentry", "0336_mark_deprecated_savedsearch_is_default_nullable"),
```
Comment (wedamija) [inline src/sentry/migrations/0338_add_saved_search_visibility.py:27]: Oh, no need to rebase, just this is pointing at the wrong place

Comment (evanpurkhiser) [reply src/sentry/migrations/0338_add_saved_search_visibility.py:27]: yep

**Pull request merged**